### PR TITLE
Improve tag mapping specs

### PR DESCRIPTION
**Pull request by cben**
Split from #162

- Use factory for more accurate test, matching UI-created mapping categories.
  Previous test didn't set read_only nor used 'kubernetes:' prefix,
  failed with attempted retagging implementation https://github.com/ManageIQ/manageiq/pull/16449
  and will fail with its upcoming replacement.

- Test that we don't remove user-assigned tags.

### CODE REVIEW

**Comment by cben** [comment]
@miq-bot add-label test, gaprindashvili/yes

@zeari @moolitayer Please review

**Comment by moolitayer** [comment]
@cben so this change is testing that additional tags that did not come back in refresh are not removed?
Do we always not remove tags that did not come back in refresh? if not, In which cases do we not delete them?

**Comment by cben** [comment]
@moolitayer This tests not removing a user-defined user-assigned tag.
It's assigned after 1st refresh because the object must exist for user to do Edit Tags.

We should of course be removing mapping-generated mapping-assigned tags, when the label disappears.
That will be tested in similar openshift PR I'm sending soon, where we such situation in the cassette.

**Comment by cben** [comment]
Fixed potential double tagging on 3rd refresh.
Rewrote for clearer data flow by arg passing, unrolled the loop, improved comment (I hope).
PTAL.

File: spec/models/manageiq/providers/kubernetes/container_manager/refresher_spec.rb [hunk 1]
```diff
@@ -24,40 +24,45 @@
...
+    # Volumes, PVs, and PVCs are tested in _before_deletions VCR.
   end
 
   it "will perform a full refresh on k8s" do
+    # Run three times to verify that second & third runs with existing data do not change anything
     full_refresh_test
+
+    # Now records exist, simulate user assigning tags by Edit Tags, to test later refreshes don't remove them.
+    @replicator.reload.tags |= [@user_tag]
+    @containergroup.reload.tags |= [@user_tag]
```
Comment (moolitayer) [inline spec/models/manageiq/providers/kubernetes/container_manager/refresher_spec.rb]: I might be missing something, but if the refresh does remove the user tags won't this code re set them?

Comment (cben) [reply spec/models/manageiq/providers/kubernetes/container_manager/refresher_spec.rb]: No.  This was a tricky point with the loop but is simpler now it's unrolled:
here is the one and *only* time we set the user tag.
next 2 full refreshes only perform refresh, and assert the tag is still assigned.

Do you see any clearer way to write this?  I'm really happy with the structure I reached currently, but what matters is somebody else being able to understand...

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit d2e40c479ac04aa557d7b714dbab3f6827b3cd52
Author: Mooli Tayer <mtayer@redhat.com>
Date:   Sun Nov 19 23:00:20 2017 +0200

    Merge pull request #170 from cben/tag-mapping-tests
    
    Improve tag mapping specs
    (cherry picked from commit ef6039818308c25baab8abadaf124aa28952e108)
```

**Pull request merged**